### PR TITLE
chore(demo): pagination overview tab

### LINKED ISSUE #2931: Docs: Pagination Overview
Would be very nice to add an overview tab to Pagination documentation to illustrate how to use it associated with `*ngFor=" ...  | slice"`

Most of the content to be used is already in https://github.com/ng-bootstrap/ng-bootstrap/issues/2672#issuecomment-444799743

### PR DESCRIPTION
**Pull request by benouat**
Just adding some documentation to highlight how to setup properly pagination.
Closes #2931 

---

Here is the screenshot
image

### CODE REVIEW

File: demo/src/app/components/pagination/overview/pagination-overview.component.html [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+
+  <br />
+  <ngb-alert type="warning" [dismissible]="false">
+    Be aware that both <code>page</code> and <code>pageSize</code> have default values,
+    which are respectively <code>1</code> and <code>10</code>.
+  </ngb-alert>
+
+  <h4>Filtering and sorting</h4>
+  <p>
+    If you want to add filtering or sorting on top of your pagination, we will just have to update the way we split our
```
Comment (maxokorokov) [inline demo/src/app/components/pagination/overview/pagination-overview.component.html]: If you want..., **we** will just → If you want, ... **you**'ll just ?

File: demo/src/app/components/pagination/overview/pagination-overview.component.html [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+<ngbd-overview-section [section]="sections['basic-usage']">
+  <p>
+    In order to properly operate a pagination behaviour, there are 3 important notions we have to be familiar with,
+    which are actually available as <code>@Input()</code> on the widget:
+  </p>
+  <ol>
+    <li><code>collectionSize</code> - Number of elements/items in the collection.
+      i.e. the total number of items the
+      pagination should handle.</li>
+    <li><code>pageSize</code> - Number of elements/items per page. i.e. the number
```
Comment (maxokorokov) [inline demo/src/app/components/pagination/overview/pagination-overview.component.html]: > Number of elements/items per page. i.e. the number of items each page should contain

while ok for `collectionSize`, here for `pageSize` it sounds redundant

File: demo/src/app/components/pagination/overview/pagination-overview.component.html (hunk 1 above)
Comment (benouat) [reply demo/src/app/components/pagination/overview/pagination-overview.component.html]: done, I picked **you**

File: demo/src/app/components/pagination/overview/pagination-overview.component.html (hunk 2 above)
Comment (benouat) [reply demo/src/app/components/pagination/overview/pagination-overview.component.html]: Removed the unnecessary _ i.e. the number of items each page should contain_

**Pull request merged**